lib/beaker/dsl/install_utils/foss_utils.rb

```diff
@@ -266,6 +266,12 @@ def install_puppet_on(hosts, opts={})
               if host['platform'] =~ /el-(5|6|7)/
                 relver = $1
                 install_puppet_from_rpm_on(host, opts.merge(:release => relver, :family => 'el'))
+              elsif host['platform'] =~ /sles-(11|12)/
+                relver = $1
+                # Need to get rid of whatever Puppet might have shipped on the
+                # upstream images
+                host.uninstall_package('puppet')
```

Comment:
@trevor-vaughan does the image you're using already have puppet on it, and that's why you're uninstalling it in the install method? 

@johnduarte, @cthorn42 are you guys aware of how our SLES testing images are configured? Or if not, would you point me to who is? I'd like to have them available to discuss our testing setup for this platform internally.